File: TaboolaSDKExampleObjective-C/TaboolaSDKExampleObjective-C/ClassicMainScrollViewController.m
//
//  ClassicMainScrollViewController.m
//  TaboolaSDKExampleObjective-C
//
//  Created by Sasa Jovanovic on 15.1.25..
//

#import "ClassicMainScrollViewController.h"
#import "ClassicFrameMetaAdController.h"
#import "ClassicFrameScrollViewController.h"
#import "ClassicConstraintsMetaAdViewController.h"
#import "ClassicConstraintsScrollViewController.h"

@implementation ClassicMainScrollViewController

// MARK: - View life cycle
- (void)viewDidLoad {
    [super viewDidLoad];
}

// MARK: - Actions
- (IBAction)didTapFrames:(UIButton *)sender {
    if (self.isForMetaAds) { // Go to Taboola with Meta integration screen
        ClassicFrameMetaAdController *viewController = [ClassicFrameMetaAdController new];
        [self.navigationController pushViewController:viewController animated:YES];
    } else { // Go to Taboola only screen
        UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
        ClassicFrameScrollViewController *viewController = [storyboard instantiateViewControllerWithIdentifier:@"ClassicFrameScrollViewController"];
        [self.navigationController pushViewController:viewController animated:YES];
    }
}

- (IBAction)didTapConstraints:(UIButton *)sender {
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    
    if (self.isForMetaAds) { // Go to Taboola with Meta integration screen
        ClassicConstraintsMetaAdViewController *viewController = [storyboard instantiateViewControllerWithIdentifier:@"ClassicConstraintsMetaAdViewController"];
        [self.navigationController pushViewController:viewController animated:YES];
    } else { // Go to Taboola only screen
        ClassicConstraintsScrollViewController *viewController = [storyboard instantiateViewControllerWithIdentifier:@"ClassicConstraintsScrollViewController"];
        [self.navigationController pushViewController:viewController animated:YES];
    }
}

@end

File: TaboolaSDKExampleSwift/TaboolaSDKExampleSwift/ClassicMainScrollViewController.swift
//
//  ClassicMainScrollViewController.swift
//  TaboolaSDKExampleSwift
//
//  Created by Sasa Jovanovic on 14.1.25..
//

import UIKit

class ClassicMainScrollViewController: UIViewController {
    
    var isForMetaAds:Bool = false

    //     MARK: - View life cycle
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    //     MARK: - Actions
    @IBAction func didTapFrames(_ sender: UIButton) {
        if isForMetaAds { // Go to Taboola with Meta integration screen
            let framesMetaAdController = ClassicFrameMetaAdController()
            self.navigationController?.pushViewController(framesMetaAdController, animated: true)
        } else { // Go to Taboola only screen
            let storyboard = UIStoryboard(name: "Main", bundle: nil)
            let viewController = storyboard.instantiateViewController(withIdentifier: "ClassicFrameScrollViewController") as? ClassicFrameScrollViewController
            navigationController?.pushViewController(viewController!, animated: true)
        }
    }
    
    @IBAction func didTapConstraints(_ sender: UIButton) {
        let storyboard = UIStoryboard(name: "Main", bundle: nil)
        if isForMetaAds { // Go to Taboola with Meta integration screen
            let viewController = storyboard.instantiateViewController(withIdentifier: "ClassicConstraintsMetaAdViewController") as? ClassicConstraintsMetaAdViewController
            navigationController?.pushViewController(viewController!, animated: true)
        } else { // Go to Taboola only screen
            let viewController = storyboard.instantiateViewController(withIdentifier: "ClassicConstraintsScrollViewController") as? ClassicConstraintsScrollViewController
            navigationController?.pushViewController(viewController!, animated: true)
            
        }
    }
    
}

